spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java

```diff
@@ -151,4 +161,25 @@ TableCatalog tableCatalog() {
       return tableCatalog;
     }
   }
+
+  /**
+   * Closes the procedure's executor service if a new one was created.
+   */
+  protected void closeService() {
+    if (executorService != null) {
+      executorService.shutdown();
+    }
+  }
+
+  protected ExecutorService executorService(int concurrentDeletes, String nameFormat) {
+    Preconditions.checkArgument(executorService == null, "Cannot create a new executor service, one already exists.");
+    executorService = MoreExecutors.getExitingExecutorService(
+        (ThreadPoolExecutor) Executors.newFixedThreadPool(
+            concurrentDeletes,
+            new ThreadFactoryBuilder()
+                .setNameFormat(nameFormat + "-%d")
```

Comment:
give `nameFormat` a default? Or cannot be null via a `Preconditions`?

Reply:
Sure I can protect against that, this is an internal method so i'm not that worried.

Reply:
Agreed that it's not too much of a worry. Based on the usage, this is all very much internal. So somebody would have to explicitly pass in `null` as the argument. But a `Preconditions` check isn't a bad idea.